### ISSUE TITLE
slow launcher

### ISSUE THREAD

**Issue by atar-axis** [pop-os/shell#1289]
**(1) Issue/Bug Description:**

when i open the launcher and start typing something, like `fire` for e.g. `firefox` it takes around 3 seconds to load the results, which is quite long in my eyes

**(2) Steps to reproduce (if you know):**

1. launch launcher using OS key
2. type `fire`

**(3) Expected behavior:**

faster loading of the results

**(4) Distribution (run `cat /etc/os-release`):**
 ``
NAME="Pop!_OS"
VERSION="21.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 21.10"
VERSION_ID="21.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=impish
UBUNTU_CODENAME=impish
LOGO=distributor-logo-pop-os
``

**(5) Gnome Shell version:**

see below

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**

```
pop-shell:
  Installed: 1.1.0~1639174177~21.10~de901ae
  Candidate: 1.1.0~1639174177~21.10~de901ae
  Version table:
 *** 1.1.0~1639174177~21.10~de901ae 1001
       1001 http://apt.pop-os.org/release impish/main amd64 Packages
       1001 http://apt.pop-os.org/release impish/main i386 Packages
        100 /var/lib/dpkg/status
``


**(7) Where was Pop Shell installed from:**

preinstalled


**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**

nothing special

**(9) Other Installed/Enabled Extensions:**

nothing special

**(10) Other Notes:**

nothing special

**Comment by mmstick** [comment]
I have a Raspberry Pi 400 which completes the search in a fraction of a second, so this sounds strange. Perhaps Any messages printed in `journalctl -fk` when doing this search? What file system are you using?

**Comment by thiagoa** [comment]
I guess it is the same problem I have on all my computers (with different hardware). The launcher gets slow over time during long-running sessions. I've been using it less and less to avoid this problem, but it always comes to a point where it's sufficiently laggy and compromises UX.

Today I woke up my laptop and it was very laggy, which forced me to restart GNOME. I use Wayland (which doesn't support the `r` command) so it's a bit inconvenient to restart my session. After that, it got speedy again. My hunch is that the problem is with the JavaScript, not the backend. Surprising that the folks at System76 don't seem to have this problem, as it's been there since the transition to a Rust backend.

**Comment by mmstick** [comment]
I use it frequently but haven't noticed any issues still. Do you also get this issue in X11? As I don't use Wayland.

**Comment by thiagoa** [comment]
@mmstick Yep, both in X11 and Wayland. To reproduce the problem, you can:

1. Fire the launcher and type random strings
2. Dismiss the launcher
3. Repeat 1 and 2 long enough for a few minutes, and you will see it getting more and more sluggish; results are shown synchronously instead of asynchronously (in other words, the typed character only appears when there are results for it, whereas in tools like Ulauncher the results are fetched asynchronously, which doesn't block the characters from showing up)

I get to that point almost daily with normal use, and since I don't turn off my laptop, it's a very normal issue for me.

**Comment by thiagoa** [comment]
@mmstick I've edited the steps to be more accurate. Thanks!

**Comment by mmstick** [comment]
I wonder if there's a specific search result causing it. I'm still not able to replicate from my end with these steps. Maybe a few minutes aren't long enough. That said, I don't doubt that there's performance issues and possible memory leaks with some gobjects managed by the JS code.
